velox/common/file/File.cpp

```diff
@@ -238,6 +246,21 @@ uint64_t LocalReadFile::preadv(
   return totalBytesRead;
 }
 
+folly::SemiFuture<uint64_t> LocalReadFile::preadvAsync(
+    uint64_t offset,
+    const std::vector<folly::Range<char*>>& buffers) const {
+  auto promise = std::make_unique<folly::Promise<uint64_t>>();
+  folly::SemiFuture<uint64_t> future = promise->getSemiFuture();
+  executor_->add([this,
+                  _promise = std::move(promise),
+                  _offset = offset,
+                  _buffers = buffers]() {
+    auto delegateFuture = ReadFile::preadvAsync(_offset, _buffers);
+    _promise->setValue(delegateFuture.wait().value());
+  });
+  return future;
```

Comment:
The local file system shutdown or dtor needs to wait for the executor to shutdown first? Thanks!

Reply:
The ThreadPoolExecutor dtor will let join keep alive https://github.com/facebook/folly/blob/main/folly/executors/ThreadPoolExecutor.cpp#L63, so local file system should wait for the executor to shutdown first.